packages/core-app-api/src/apis/implementations/auth/guest/GuestAuth.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthRequestOptions,
+  BackstageIdentityApi,
+  ProfileInfo,
+  ProfileInfoApi,
+  SessionApi,
+  SessionState,
+  BackstageIdentityResponse,
+} from '@backstage/core-plugin-api';
+import { Observable } from '@backstage/types';
+import { RefreshingAuthSessionManager } from '../../../../lib/AuthSessionManager';
+import { SessionManager } from '../../../../lib/AuthSessionManager/types';
+import { AuthApiCreateOptions } from '../types';
+import { RefreshingDirectAuthConnector } from '../../../../lib/AuthConnector/RefreshingDirectAuthConnector';
+
+type GuestSession = {
+  profile: ProfileInfo;
+  backstageIdentity: BackstageIdentityResponse;
+};
+
+const DEFAULT_PROVIDER = {
+  id: 'guest',
+  title: 'Guest',
+  icon: () => null,
+};
+
+/**
+ * Implements a guest auth flow. Heavily based on SAML flow with added support for refreshing the token.
+ *
+ * @public
+ */
+export default class GuestAuth
```

Comment:
I think you could use this instead right? https://github.com/backstage/backstage/blob/b5c65ca6e9ec7f192ddbf59afc20f1096e8c724e/packages/core-components/src/layout/ProxiedSignInPage/ProxiedSignInIdentity.ts#L80